From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 2 Sep 2020 09:26:00 +0300
Subject: [PATCH] Fix vanilla generation deadlock

Comment from ivan: expect this to be dropped when paper ports this

diff --git a/src/main/java/net/minecraft/server/RegionLimitedWorldAccess.java b/src/main/java/net/minecraft/server/RegionLimitedWorldAccess.java
index e9fbeaf309735ea5fcb0c53b8b2485b0c5dd43be..23341b06b3a2bc98829520045308e006afe29f21 100644
--- a/src/main/java/net/minecraft/server/RegionLimitedWorldAccess.java
+++ b/src/main/java/net/minecraft/server/RegionLimitedWorldAccess.java
@@ -379,6 +379,6 @@ public class RegionLimitedWorldAccess implements GeneratorAccessSeed {
 
     @Override
     public Stream<? extends StructureStart<?>> a(SectionPosition sectionposition, StructureGenerator<?> structuregenerator) {
-        return this.f.a(sectionposition, structuregenerator);
+        return this.f.getStructureManager().a(this).a(sectionposition, structuregenerator); // Yatopia
     }
 }
